pkg/test/docker.go

```diff
@@ -43,23 +44,25 @@ func newDockerRunner() *dockerRunner {
 
 	pool.MaxWait = 2 * time.Minute
 
-	resources := make([]*dockertest.Resource, 0)
+	containers := make([]string, 0)
```

Comment:
is there any reason not to save a pointer to a dockertest.Resource? i find this quite nice when debugging to be able to see all properties of a container. Especially when adding func functionality in the integration test of course

Reply:
By using the container name we can add it to the list even before starting the container. In other words: we are 100% sure that if we started container X that X will be in the list of containers (used later for cleanup).

And if we use the name instead of the resource, then we do not need to save them anymore. If you need it for debugging then you can save them temporarily in a variable.

Reply:
If the container did not start, then there is no need to try and clean it up. Do i not see something here?

Reply:
never mind, just had a look at dockertest. Due to the way they implemented `RunWithOptions` there is no better way.